src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionAddActionTest.java

```diff
@@ -470,6 +474,54 @@ public void testExecuteAndPostProcessConstSumOption() {
                                     + "question:</span> Split points among the options."
                                     + "|||/page/instructorFeedbackQuestionAdd";
         AssertHelper.assertLogMessageEquals(expectedLogMessage, action.getLogMessage());
+
+        ______TS("add min and max points constraint -> add failure ");
```

Comment:
Test case missing:
1. Only has `FEEDBACK_QUESTION_CONSTSUMPOINTS_HAS_MINPOINTS_CONSTRAINT` and < 0
2. Only has `FEEDBACK_QUESTION_CONSTSUMPOINTS_HAS_MINPOINTS_CONSTRAINT` and < 0

The thing is that if u can write comprehensive test cases at this testing level (integration test), then there is **NO NEED** to write equivalent tests at end-to-end test level. If u ask me whether we can write comprehensive test cases at end-to-end tests instead, I would say no. You will see the benefits of writing test cases at integration level as follows.
1. It's much easier and faster to run integration tests than end-to-end tests
2. It's much faster to debug things at integration test level (Assertions with comparison for HTML files are very hard to debug).